### PR TITLE
Disambiguate a case where multiple test suites have the same name

### PR DESCRIPTION
**Pull request by Vlatombe**
This causes test case resolution to be ambiguous if resolved from the
wrong test suite.

The consequence is for a failing test case that its age could reset
unexpectedly.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by timja — APPROVED** [review]
I think okay, @jglick any thoughts?

File: src/test/java/hudson/tasks/junit/pipeline/JUnitResultsStepTest.java [hunk 1]
```diff
@@ -385,14 +368,57 @@ public void skipBuildUnstable() throws Exception {
...
+        CaseResult caseResult = findCaseResult(r, "aClass.methodName");
+        assertNotNull(caseResult);
+        assertEquals(1, caseResult.getAge());
+
+        // Run a second build, age should increase
+        r = rule.waitForCompletion(j.scheduleBuild2(0).waitForStart());
+        rule.assertBuildStatus(Result.UNSTABLE, r);
+        caseResult = findCaseResult(r, "aClass.methodName");
+        assertNotNull(caseResult);
+        assertEquals(2, caseResult.getAge());
```
Comment (jglick) [inline src/test/java/hudson/tasks/junit/pipeline/JUnitResultsStepTest.java:403]: Is this what fails with `src/main/` reverted? Not obvious to me how the test case is verifying that there are two suites.

**Review by jglick — COMMENTED** [review]
No particular opinion. Is there a motivation for this?

**Comment by jglick** [comment]
IIUC this is a variant of https://issues.jenkins.io/browse/JENKINS-30413?

**Review by jglick — APPROVED** [review]
OK I think. As always there is a high risk of regression in this plugin since it is so widely used and there are so many edge cases.

**Comment by leonardokochake** [comment]
After this update our test report return `NullPointerException` with the following stackTraceError:

```
java.lang.NullPointerException
	at hudson.tasks.junit.TestResult.getCase(TestResult.java:931)
	at hudson.tasks.junit.CaseResult.getPreviousResult(CaseResult.java:540)
	at hudson.tasks.junit.CaseResult.recomputeFailedSinceIfNeeded(CaseResult.java:453)
	at hudson.tasks.junit.CaseResult.freeze(CaseResult.java:704)
	at hudson.tasks.junit.SuiteResult.freeze(SuiteResult.java:454)
	at hudson.tasks.junit.TestResult.freeze(TestResult.java:825)
	at hudson.tasks.junit.TestResultAction.load(TestResultAction.java:238)
	at hudson.tasks.junit.TestResultAction.getResult(TestResultAction.java:157)
	at hudson.tasks.junit.TestResultAction.getResult(TestResultAction.java:66)
	at hudson.tasks.test.AbstractTestResultAction.findCorrespondingResult(AbstractTestResultAction.java:259)
	at hudson.tasks.test.TestResult.getPreviousResult(TestResult.java:151)
	at hudson.tasks.junit.TestResult.getPreviousResult(TestResult.java:234)
	at hudson.tasks.junit.CaseResult.getPreviousResult(CaseResult.java:536)
	at hudson.tasks.junit.CaseResult.recomputeFailedSinceIfNeeded(CaseResult.java:453)
	at hudson.tasks.junit.CaseResult.freeze(CaseResult.java:704)
	at hudson.tasks.junit.SuiteResult.freeze(SuiteResult.java:454)
	at hudson.tasks.junit.TestResult.freeze(TestResult.java:825)
	at hudson.tasks.junit.TestResultAction.load(TestResultAction.java:238)
	at hudson.tasks.junit.TestResultAction.getResult(TestResultAction.java:157)
	at hudson.tasks.junit.TestResultAction.mergeResult(TestResultAction.java:294)
	at org.jenkinsci.plugins.xunit.XUnitProcessor.recordTestResult(XUnitProcessor.java:357)
	at org.jenkinsci.plugins.xunit.XUnitProcessor.process(XUnitProcessor.java:161)
	at org.jenkinsci.plugins.xunit.pipeline.XUnitResultsStepExecution.run(XUnitResultsStepExecution.java:82)
	at org.jenkinsci.plugins.xunit.pipeline.XUnitResultsStepExecution.run(XUnitResultsStepExecution.java:48)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

For now I revert the plugin to version 1.56 and tests are back being parsed correctly.
Can someone take a look?

Thanks
